extensions/spring-xd-extension-reactor/src/main/java/org/springframework/xd/integration/reactor/net/ReactorPeerFactoryBean.java

```diff
@@ -0,0 +1,280 @@
+	 * @return {@literal this}
+	 */
+	public ReactorPeerFactoryBean<IN, OUT> setHost(String host) {
+		Assert.notNull(host, "Host cannot be null.");
+		this.host = host;
+		return this;
+	}
+
+	/**
+	 * Set the port to which this server will bind.
+	 *
+	 * @param port the port to bind to (defaults to {@code 3000})
+	 * @return {@literal this}
+	 */
+	public ReactorPeerFactoryBean<IN, OUT> setPort(int port) {
+		Assert.isTrue(port > 0, "Port must be greater than 0");
+		this.port = port;
+		return this;
+	}
+
+	/**
+	 * Set the type of {@link reactor.io.codec.Codec} to use to managing encoding and decoding of the data. <p> The
+	 * default options for codecs are: <ul> <li>{@code bytes} - Use the standard byte array codec.</li> <li>{@code
+	 * string} - Use the standard String codec.</li> <li>{@code syslog} - Use the standard Syslog codec.</li> </ul>
+	 * </p>
+	 *
+	 * @param codec the codec
+	 * @return {@literal this}
+	 */
+	public ReactorPeerFactoryBean<IN, OUT> setCodec(String codec) {
+		this.delegateCodec = codecs.get(codec);
+		Assert.notNull(delegateCodec, "No Codec found for type " + codec);
+		return this;
+	}
+
+	/**
+	 * Set the type of framing to use. <p> The options for framing are: <ul> <li>{@code linefeed} - Means use an
+	 * LF-delimited linefeed codec.</li> <li>{@code length} - Means use a length-field based codec where the initial
+	 * bytes of a message are the length of the rest of the message.</li> </ul> </p>
+	 *
+	 * @param framing type of framing
+	 * @return {@literal this}
+	 */
+	public ReactorPeerFactoryBean<IN, OUT> setFraming(String framing) {
+		Assert.isTrue("linefeed".equals(framing) || "length".equals(framing));
+		this.framing = framing;
+		return this;
+	}
+
+	/**
+	 * Set the length of the length field if using length-field framing.
+	 *
+	 * @param lengthFieldLength {@code 2} for a {@code short}, {@code 4} for an {@code int} (the default), or {@code 8}
+	 *                          for a {@code long}
+	 * @return {@literal this}
+	 */
+	public ReactorPeerFactoryBean<IN, OUT> setLengthFieldLength(int lengthFieldLength) {
+		this.lengthFieldLength = lengthFieldLength;
+		return this;
+	}
+
+
+	@Override
+	public ReactorPeer<IN, OUT, ?> getObject() throws Exception {
+
+		if (TCP.equals(transport)) {
+			return NetStreams.tcpServer(NettyTcpServer.class, new NetStreams.TcpServerFactory<IN, OUT>() {
+				@Override
+				public Spec.TcpServerSpec<IN, OUT> apply(Spec.TcpServerSpec<IN, OUT> spec) {
+					commonSpecProperties(spec);
+					//Add specifics to TCP
+					return spec;
+				}
+			});
+		} else if (UDP.equals(transport)) {
+			return NetStreams.udpServer(NettyDatagramServer.class, new NetStreams.UdpServerFactory<IN, OUT>() {
+				@Override
+				public Spec.DatagramServerSpec<IN, OUT> apply(Spec.DatagramServerSpec<IN, OUT> spec) {
+					commonSpecProperties(spec);
+					//Add specifics to UDP
+					return spec;
+				}
+			});
+		} else if (HTTP.equals(transport)) {
```

Comment:
good question, although you can stream over http (which is itself based on tcp).